packages/studio/src/components/IFramePortal.tsx

```diff
@@ -1,27 +1,50 @@
 import { createPortal } from "react-dom";
-import { Dispatch, PropsWithChildren, SetStateAction, useEffect } from "react";
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  useEffect,
+  useRef,
+} from "react";
+import useStudioStore from "../store/useStudioStore";
+import { EXPANDED_VIEWPORTS } from "./Viewport/defaults";
+import { twMerge } from "tailwind-merge";
 
 export default function IFramePortal(
   props: PropsWithChildren<{
-    className?: string;
     title: string;
     iframeEl: HTMLIFrameElement | null;
     setIframeEl: Dispatch<SetStateAction<HTMLIFrameElement | null>>;
   }>
 ) {
+  const previewRef = useRef<HTMLDivElement>(null);
   const iframeDocument = props.iframeEl?.contentWindow?.document;
+  const [viewportDimensions] = useStudioStore((store) => [
+    store.pagePreview.viewportDimensions,
+  ]);
+  const dimensions =
+    viewportDimensions.width * (previewRef.current?.clientHeight ?? 0) >
+    viewportDimensions.height * (previewRef.current?.clientWidth ?? 0)
```

Comment:
For my own edification, could you give me a bit more detail on this logic? At a high-level I want to understand when we need to switch between w-full and h-full.

Reply:
So it's basically: if the ratio (width / height) of the viewport is greater than the ratio of the preview div then it should use w-full. If it instead used h-full, it would overflow from the div. 
This is because the aspect ratio works best if it has some style letting it know if height or width have pre-set dimensions. 

I rearranged the func to use * instead of / though.